### ISSUE TITLE
Method-obscured-issue for property.setter when property type is child class

### ISSUE THREAD

**Issue by lumpi101** [microsoft/pylance-release#1683]
<h3> Environment data </h3>

-   Language Server version: 2021.8.1
-   OS and version: linux x64 (via WSL)
-   Python version (and distribution if applicable, e.g. Anaconda): 3.10.0rc1 (via pyenv)
-   python.analysis.indexing: undefined
-   python.analysis.typeCheckingMode: basic

<h3> Expected behaviour </h3>

no complaint

<h3> Actual behaviour </h3>

pylance complains: `Method declaration "foo" is obscured by a declaration of the same name` and references to the property setter definition

<h3> Code Snippet / Additional information </h3>

```python
from __future__ import annotations
from abc import ABC

class MyBase(ABC):
    def __init__(self, foo: MyChildA | None) -> None:
        self._foo = foo
    
    @property
    def foo(self) -> MyChildA: # pylance: Method declaration "foo" is obscured by a declaration of the same name
        if self._foo is None:
            raise RuntimeError
        return self._foo
    
    @foo.setter
    def foo(self, value: MyChildA):
        self._foo = value

class MyChildA(MyBase):
    pass
```

The pylance issue disappears either when removing the ABC inheritance or when the (foo-)property return type is declared to be MyBase but none of both work-arounds actually suits me.

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by heejaechang** [comment]
it should be fixed in 2021.8.2 released today (2021.8.19)
